### DISCUSSION TITLE
Custom callback URL

### DISCUSSION THREAD

**Discussion by dcods22** [nextauthjs/next-auth#8517] [Help]
### Question 💬

I am trying to configure our applications auth which uses okta.  My auth works fine locally because it is on the localhost domain, but when deploying our application we deploy to kubernetes then add fastly CDN on top of our kubernetes application so we can have a single domain map to this NextJS application and others in the future.  When we try to login from the fastly domain, NextAuth sets the callback URL to the domain of the Kubernetes application not the fastly CDN and it also sends the kubernetes domain as the redirect_uri in the authorize call to OKTA.  I want to be able to set the domain of all of the NextAuth calls to be my fastly domain.  I have the fastly domain as the value of `NEXTAUTH_URL` so I figured this would override the domain for these various calls but it does not.  I found [this issue](https://github.com/nextauthjs/next-auth/pull/7038#issuecomment-1482110414) which says you want to control it? But why? This does not allow you to appy any sort of DNS changes on top of your server this way.  

How can I accomplish this with the current configurations of NextAuth?

### How to reproduce ☕️

Not easily reproducable

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by dcods22** [comment]
Also I am running into an issue where the cookies are not on the fastly domain and since NextAuth relies on these cookies to check the sessions, it does not have access on both applications. I want to be able to put any domain on top of my application and not have to worry about what behind the scenes is running

**Comment by vwallen** [comment]
I'm running into a similar issue trying to use AuthJS on a Nuxt project hosted on Firebase. The `redirect_url` is being set to the URL of the cloud function rendering the site rather than the URL used to access the site. If this isn't something that can be made configurable, then AuthJS is essentially useless to me (and apparently others).

**Comment by tao1** [comment]
I confirm. nextauth is useless if we cannot customize callback url.

**Comment by kanekv** [comment]
Pretty much any oauth lib allows to provide custom callback url. What is this limitation for? What's the purpose other than make it harder to use?

**Comment by devhmac** [comment]
Why is there dead silence on this one? It seems like a no brainer to allow this, and there is no real explained rationale behind locking this down when any other standard auth library would allow the user to override it _at least_ on a custom provider.

A bunch of followup on [this issue](https://github.com/nextauthjs/next-auth/pull/7038) asking for options or clarification and it was just closed with no real response.
